src/Util/StringUtil/StringUtil.js

```diff
@@ -14,6 +14,39 @@ class StringUtil {
 
     return text.replace(urlRegex, `<a href="$1" target="_blank">$1</a>`);
   }
+
+  /**
+   * This coereces the value of a string by casting it to the most plausible
+   * datatype, guessed by the value itself.
+   *
+   * @param {String} string The input string to coerce.
+   * @return {string|boolean|float|object} The coerced value.
+   */
+  static coerce(string) {
+    if (!(string instanceof String || typeof string === 'string')) {
+      return string;
+    }
+
+    if (string.toLowerCase() === 'true') {
+      return true;
+    } else if (string.toLowerCase() === 'false') {
+      return false;
+    } else if (/^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$/.test(string)) {
+      return parseFloat(string);
+    } else if (string.startsWith('[') && !string.startsWith('[[')) {
```

Comment:
Why wouldn't we coerce `[[123,345]]`?

Reply:
I think this was some special BfS stuff. I'll remove it.